### PR TITLE
Add Beacon transport type

### PR DESCRIPTION
**Pull request by tomasdev**
This is very poorly documented so it has been missed.

Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:

- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:

- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the `package.json`.

If removing a declaration:

- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/gtag.js/index.d.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 07483c7b6f633daa46daa3137021f6f0c21cd79f
</p>

**Comment by jakebailey** [comment]
The link you have in the PR goes to an error page, so this can't really be confirmed.

**Comment by tomasdev** [comment]
Uh? Are you behind some sort of VPN?
Screenshot 2024-05-06 at 3 20 33 PM

**Comment by jakebailey** [comment]
Must have been a temporary problem, weird.

File: types/gtag.js/index.d.ts [hunk 1]
```diff
@@ -97,6 +97,8 @@ declare namespace Gtag {
         value?: number | undefined;
         event_label?: string | undefined;
         event_category?: string | undefined;
+        /** @see https://developers.google.com/analytics/devguides/collection/gtagjs/sending-data#specify_different_transport_mechanisms */
+        transport_type?: 'beacon' | undefined;
```
Comment (jakebailey) [inline types/gtag.js/index.d.ts:101]: The docs imply this could be `image` or `xhr`, right?

**Review by sandersn — CHANGES_REQUESTED** [review]
Requesting changes so the bot knows the current status.

**Comment by tomasdev** [comment]
Done!

**Review by sandersn — APPROVED** [review]

**Comment by KsAkira10** [comment]
LGTM!

**Pull request merged**